src/util/bitvector_expr.h

```diff
@@ -707,6 +707,10 @@ class binary_overflow_exprt : public binary_predicate_exprt
   binary_overflow_exprt(exprt _lhs, const irep_idt &kind, exprt _rhs)
     : binary_predicate_exprt(std::move(_lhs), make_id(kind), std::move(_rhs))
   {
+    INVARIANT(
```

Comment:
I quite like this.

I favour things being correct by construction, and subsequently, for the invariants to be set as early as possible.